### PR TITLE
fix(FEC-11791): share url has cross-site scripting vulnerability

### PR DESCRIPTION
**Pull request by lianbenjamin**
**description of the issue:**
when share plugin is enabled for the player, the share url is not being sanitized, which exposes a security vulnerability.

**the solution:**
sanitizing encoded and decoded share url.

Solves FEC-11791

### CODE REVIEW

File: modules/KalturaSupport/components/share/share.js [hunk 1]
```diff
@@ -616,7 +616,7 @@
 					res = document.URL;
 				}
 			}
-			return res;
+			return kWidget.sanitize(decodeURIComponent(res));
```
Comment (yairans) [inline modules/KalturaSupport/components/share/share.js:619]: do we need to encode the url again?

Comment (lianbenjamin) [reply modules/KalturaSupport/components/share/share.js:619]: no, the decodeURIComponent replaces each escape sequence in the encoded with the character that it represents in ascii. if it's not an escaping char, it moves on. for example:
https://www.kaltura.com/.../embed/dynamic?%22%3E%3Cimg%20src=x%20onerror=prompt(4)%3E
would turn into:
https://www.kaltura.com/.../embed/dynamic?">
by doing so, the sanitizer would be able to replace any escaping chars with empty strings. without that decoding stage, sanitizer won't work in cases where there is encoded uri. if the uri is not encoded (most cases), the decoder does nothing and returns the uri string as-is.

**Comment by yairans** [comment]
@OrenMe please take a look on this PR

File: modules/KalturaSupport/components/share/share.js (hunk 1 above)
Comment (yairans) [reply modules/KalturaSupport/components/share/share.js:619]: i think encode it again it safer. if it was encoded we have to keep it encoded. if not we just encode the url, what the side effect? @OrenMe wdyt?

**Review by yairans — APPROVED** [review]

**Pull request merged**